R/summary.R

```diff
@@ -321,431 +284,139 @@ summary.gs_design <- function(object,
+  col_vars <- names(col_decimals)
 
-  if ("nominal p" %in% x_decimals$col_vars) {
-    x_decimals <- x_decimals %>% mutate(col_vars = dplyr::if_else(col_vars == "nominal p", "Nominal p", col_vars))
-  }
+  output <- output %>% group_by(Analysis) %>% select(all_of(col_vars))
+  # Set the decimals to display ----
+  for (j in col_vars) output[[j]] <- round2(output[[j]], col_decimals[j])
 
+  output <- add_class(
+    output, method, intersect("non_binding", class(object)), method, "gs_design"
+  )
 
-  output <- output %>% select(x_decimals$col_vars)
-  if ("Z" %in% colnames(output)) {
-    output <- output %>% dplyr::mutate_at(
-      "Z",
-      round,
-      (x_decimals %>% filter(col_vars == "Z"))$col_decimals
-    )
-  }
-  if ("~HR at bound" %in% colnames(output)) {
-    output <- output %>% dplyr::mutate_at(
-      "~HR at bound",
-      round,
-      (x_decimals %>% filter(col_vars == "~HR at bound"))$col_decimals
-    )
-  }
-  if ("~Risk difference at bound" %in% colnames(output)) {
-    output <- output %>% dplyr::mutate_at(
-      "~Risk difference at bound",
-      round,
-      (x_decimals %>% filter(col_vars == "~Risk difference at bound"))$col_decimals
-    )
-  }
-  if ("Nominal p" %in% colnames(output)) {
-    output <- output %>% dplyr::mutate_at(
-      "Nominal p",
-      round,
-      (x_decimals %>% filter(col_vars == "Nominal p"))$col_decimals
-    )
-  }
-  if ("Alternate hypothesis" %in% colnames(output)) {
-    output <- output %>% dplyr::mutate_at(
-      "Alternate hypothesis",
-      round,
-      (x_decimals %>% filter(col_vars == "Alternate hypothesis"))$col_decimals
-    )
-  }
-  if ("Null hypothesis" %in% colnames(output) && is.vector(output[["Null hypothesis"]], mode = "numeric")) {
-    output <- output %>% dplyr::mutate_at(
-      "Null hypothesis",
-      round,
-      (x_decimals %>% filter(col_vars == "Null hypothesis"))$col_decimals
-    )
+  # Save the full alpha as an attribute of the output summary table
+  # Use input$alpha when given power to calculate sample size
+  attr(output, "full_alpha") <- object$input$alpha %||% {
+    # when given sample size to calculate power
+    if (is.list(object$input$upar)) object$input$upar$total_spend else 0.025
   }
 
-  class(output) <- c(method, "gs_design", class(output))
-  if ("non_binding" %in% class(object)) {
-    class(output) <- c("non_binding", class(output))
-  }
+  return(output)
+}
 
-  # Save the full alpha as an attribute of the output summary table
-  attr(output, "full_alpha") <-
-    if (is.null(object$input$alpha)) {
-      # case when given sample size to calculate power
-      if (!is.list(object$input$upar)) {
-        0.025
-      } else {
-        object$input$upar$total_spend
-      }
-    } else {
-      # case when given power to calcuate sample size
-      object$input$alpha
-    }
+# get a named vector of decimals (names are variable names)
+get_decimals <- function(vars, decs, vars_default, decs_default) {
```

Comment:
This function should be equivalent to https://github.com/Merck/gsDesign2/blob/7a6ebb05c260926e781e127315d08cbf03449ab9/R/summary.R#L428-L462 but my code may be a little too abstract to understand.

Reply:
My code was very "brute force", so I'm fine with it being replaced with your more elegant function. I wrote a ton of tests for the various combinations of user input arguments, and they all pass, so I am confident it is equivalent.

Reply:
I can't praise the tests in this package enough. I have been quite impressed by their coverage.